src/overlays/actors/ovl_Obj_Tsubo/z_obj_tsubo.c

```diff
@@ -33,79 +53,639 @@ const ActorInit Obj_Tsubo_InitVars = {
+    s32 pad2;
+    s32 acHit = (this->cylinderCollider.base.acFlags & AC_HIT) != 0;
+
+    if (acHit) {
+        this->cylinderCollider.base.acFlags &= ~AC_HIT;
+    }
+    if (this->cylinderCollider.base.ocFlags1 & OC1_HIT) {
+        this->cylinderCollider.base.ocFlags1 &= ~OC1_HIT;
+    }
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        this->actor.room = -1;
+        this->actor.flags |= ACTOR_FLAG_10;
+        if ((type != OBJ_TSUBO_TYPE_3) && func_800A817C(OBJ_TSUBO_P003F(&this->actor))) {
+            func_80927690(this, globalCtx);
+        }
+        func_80927818(this, globalCtx, 0);
+        //! @bug: This function should only pass Player*: it uses *(this + 0x153), which is meant to be
+        //! player->currentMask, but in this case is garbage in the collider
+        func_800B8E58((Player*)&this->actor, NA_SE_PL_PULL_UP_POT);
+        func_80928D6C(this);
+    } else if ((this->unk_19B != 0) ||
+               (acHit && (this->cylinderCollider.info.acHitInfo->toucher.dmgFlags & 0x058BFFBC))) {
+        typeData = &sPotTypeData[type];
+        this->unk_19B = 0;
+        if ((this->actor.bgCheckFlags & 0x20) && (this->actor.depthInWater > 15.0f)) {
+            typeData->breakPot3(this, globalCtx);
+        } else {
+            typeData->breakPot1(this, globalCtx);
+        }
+        if (type == OBJ_TSUBO_TYPE_3) {
+            func_8092762C(this, globalCtx);
+        } else {
+            func_80927690(this, globalCtx);
+        }
+        func_80927818(this, globalCtx, 1);
+        SoundSource_PlaySfxAtFixedWorldPos(globalCtx, &this->actor.world.pos, 20, NA_SE_EV_POT_BROKEN);
+        if (ObjTsubo_IsSceneNotGohtOrTwinmold(this, globalCtx)) {
+            Actor_MarkForDeath(&this->actor);
+        } else {
+            func_809291DC(this);
+        }
+    } else {
+        if (!this->unk_195) {
+            Actor_MoveWithGravity(&this->actor);
+            Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 15.0f, 15.0f, 0.0f, 0x44);
+            if ((this->actor.bgCheckFlags & 1) &&
+                DynaPoly_GetActor(&globalCtx->colCtx, this->actor.floorBgId) == NULL) {
+                this->unk_195 = true;
+                this->actor.flags &= ~ACTOR_FLAG_10;
+            }
+        }
+        if ((this->actor.xzDistToPlayer < 800.0f) || (gSaveContext.save.entranceIndex == 0xD010)) {
+            Collider_UpdateCylinder(&this->actor, &this->cylinderCollider);
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->cylinderCollider.base);
+            if (this->actor.xzDistToPlayer < 150.0f) {
+                CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->cylinderCollider.base);
+                if (this->actor.xzDistToPlayer < 100.0f) {
+                    s16 yawDiff = this->actor.yawTowardsPlayer - GET_PLAYER(globalCtx)->actor.world.rot.y;
+                    s32 absYawDiff = ABS_ALT(yawDiff);
+
+                    if (absYawDiff >= 0x5556) {
```

Comment:
```suggestion
                    if (absYawDiff > 0x10000 / 3) {
```

Reply:
opted for `absYawDiff > DEGF_TO_BINANG(120.0f)` instead

Reply:
I'd rather `0x10000 / 3` at least for now: there are cases in the codebase where it won't be a sensible number of degrees, like `0x10000 / 11` (and even `0x10000 / 5` is not so obvious). But I'd like some other opinions.

Reply:
At least for now I think it can be left up to the PR author since we haven't decided on the style.